app/controllers/journeys/capitalallowances/electricVehicleChargePoints/EVCPAllowanceController.scala

```diff
@@ -0,0 +1,96 @@
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import queries.Settable
+import services.SelfEmploymentService
+import services.SelfEmploymentService.clearDataFromUserAnswers
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.Logging
+import views.html.journeys.capitalallowances.electricVehicleChargePoints.EVCPAllowanceView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class EVCPAllowanceController @Inject() (override val messagesApi: MessagesApi,
+                                         navigator: CapitalAllowancesNavigator,
+                                         identify: IdentifierAction,
+                                         getData: DataRetrievalAction,
+                                         requireData: DataRequiredAction,
+                                         service: SelfEmploymentService,
+                                         formProvider: EVCPAllowanceFormProvider,
+                                         val controllerComponents: MessagesControllerComponents,
+                                         view: EVCPAllowanceView)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with Logging {
+
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val form = request.userAnswers
+        .get(EVCPAllowancePage, businessId.some)
+        .fold(formProvider(request.userType))(formProvider(request.userType).fill)
+
+      Ok(view(form, mode, request.userType, taxYear, businessId))
+  }
+
+  def onSubmit(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
+    implicit request =>
+      formProvider(request.userType)
+        .bindFromRequest()
+        .fold(
+          formErrors => Future.successful(BadRequest(view(formErrors, mode, request.userType, taxYear, businessId))),
+          answer =>
+            for {
+              (editedUserAnswers, redirectMode) <- handleGatewayQuestion(answer, request, mode, businessId)
+              updatedUserAnswers                <- service.persistAnswer(businessId, editedUserAnswers, answer, EVCPAllowancePage)
+            } yield Redirect(navigator.nextPage(EVCPAllowancePage, redirectMode, updatedUserAnswers, taxYear, businessId))
+        )
+  }
+
+  private def handleGatewayQuestion(currentAnswer: Boolean,
+                                    request: DataRequest[_],
+                                    mode: Mode,
+                                    businessId: BusinessId): Future[(UserAnswers, Mode)] = {
+    val pagesToBeCleared: List[Settable[_]] =
+      List(
+        // TODO: clear claiming allowance, useOutsideSE, selfEmploymentUSe, chargePointAmount and tax relief pages
```

Comment:
Can now clear all the pages for this controller and for the tax relief controller